packages/core/src/xr/input/XRInputManager.ts

```diff
@@ -0,0 +1,279 @@
+import { IXRInput, IXRInputEvent, IXRSession } from "@galacean/engine-design";
+import { Engine } from "../../Engine";
+import { XRController } from "./XRController";
+import { XRCamera } from "./XRCamera";
+import { XRTrackedInputDevice } from "./XRTrackedInputDevice";
+import { UpdateFlagManager } from "../../UpdateFlagManager";
+import { XRInputButton } from "./XRInputButton";
+import { XRTrackingState } from "./XRTrackingState";
+import { XRTrackedUpdateFlag } from "./XRTrackedUpdateFlag";
+import { XRTargetRayMode } from "./XRTargetRayMode";
+import { XRInputEventType } from "./XRInputEventType";
+
+/**
+ * The manager of XR input.
+ */
+export class XRInputManager {
+  /** @internal */
+  _cameras: XRCamera[] = [];
+  /** @internal */
+  _controllers: XRController[] = [];
+
+  private _engine: Engine;
+  private _inputs: IXRInput[] = [];
+  private _added: IXRInput[] = [];
+  private _updated: IXRInput[] = [];
+  private _removed: IXRInput[] = [];
+  private _statusSnapshot: XRTrackingState[] = [];
+  private _trackingUpdate: UpdateFlagManager = new UpdateFlagManager();
+
+  private _platformSession: IXRSession;
+
+  /**
+   * @internal
+   */
+  constructor(engine: Engine) {
+    this._engine = engine;
+    const { _inputs: inputs, _controllers: controllers, _cameras: cameras } = this;
+    for (let i = XRTrackedInputDevice.Length - 1; i >= 0; i--) {
+      switch (i) {
+        case XRTrackedInputDevice.Camera:
+        case XRTrackedInputDevice.LeftCamera:
+        case XRTrackedInputDevice.RightCamera:
+          cameras.push((inputs[i] = new XRCamera(i)));
+          break;
+        case XRTrackedInputDevice.Controller:
+        case XRTrackedInputDevice.LeftController:
+        case XRTrackedInputDevice.RightController:
+          controllers.push((inputs[i] = new XRController(i)));
+          break;
+        default:
+          break;
+      }
+    }
+    this._statusSnapshot.fill(XRTrackingState.NotTracking, 0, XRTrackedInputDevice.Length);
+  }
+
+  /**
+   * Returns the input instance.
+   * @param type - The input type
+   * @returns The input instance
+   */
+  getInput<T extends IXRInput>(type: XRTrackedInputDevice): T {
+    return <T>this._inputs[type];
+  }
+
+  /**
+   * Add a listener to the input update event.
+   * @param listener - The listener to add
+   */
+  addListener(listener: (type: XRTrackedUpdateFlag, param: readonly IXRInput[]) => any): void {
```

Comment:
The goal of this API is to let users know whether new XR Input devices have been removed or added?